src/library/scala/collection/convert/JavaCollectionWrappers.scala

```diff
@@ -442,8 +444,13 @@ private[collection] object JavaCollectionWrappers extends Serializable {
 
     def replace(k: K, v: V): Option[V] = Option(underlying.replace(k, v))
 
-    def replace(k: K, oldvalue: V, newvalue: V): Boolean =
-      underlying.replace(k, oldvalue, newvalue)
+    def replace(k: K, oldvalue: V, newvalue: V): Boolean = underlying.replace(k, oldvalue, newvalue)
+
+    override def lastOption: Option[(K, V)] =
+      underlying match {
+        case nav: NavigableMap[K @unchecked, V @unchecked] => Option(nav.lastEntry).map(e => (e.getKey, e.getValue))
+        case _ => Try(super.lastOption).toOption.flatten
```

Comment:
could this go as an override to `collection.concurrent.Map`?

Reply:
I will look into it.